### PR TITLE
Feat/bn128 precompile

### PR DESCRIPTION
**Pull request by niall-watr**
adds following precompiles
Bn128Add
Bn128Mul
Bn128Pairing
Blake2F

### CODE REVIEW

File: runtime/devnet/src/weights/block_weights.rs [hunk 1]
```diff
@@ -38,9 +38,15 @@ pub mod constants {
 			let w = super::constants::BlockExecutionWeight::get();
 
 			// At least 100 µs.
-			assert!(w >= 100 * constants::WEIGHT_PER_MICROS, "Weight should be at least 100 µs.");
+			assert!(
```
Comment (NachoPal) [inline runtime/devnet/src/weights/block_weights.rs:41]: To fix the weights tests compilation errors introduced by Weights V2, please follow what was done in Substrate and Cumulus: https://github.com/paritytech/cumulus/pull/1601/files

This is applicable to all weight files in this PR

**Review by NachoPal — CHANGES_REQUESTED** [review]
Follow how weight tests where fixed here: https://github.com/paritytech/cumulus/pull/1601/files

File: runtime/devnet/src/weights/block_weights.rs (hunk 1 above)
Comment (niall-watr) [reply runtime/devnet/src/weights/block_weights.rs:41]: Updated to use ref_time() in tests

**Comment by niall-watr** [comment]
@NachoPal can we get this merged into main?

**Pull request merged**